{
  "comments": [
    "Build ID: I20070323-1616\n\nI\u0027m not sure why anybody would want to do the following, discovered it while trying to find all error dialogs the detail formatter dialog could produce.\n\nWhile stopped at a breakpoint in a java program, edit a detail formatter using the context menu in the variables view.\nChange the code snippet to \"throw new Exception();\".\nWhen I pressed ok, the stack frame resumed, the variables view emptied, and the program sat there waiting.\nWhen I suspended the main thread, it returned to where the breakpoint was.\n\nI tried it again using \u0027throw new NullPointerException();\u0027 and didn\u0027t see the same behaviour, but when I terminated the target, I got the following exception in the log...\n\neclipse.buildId\u003dI20070323-1616\njava.version\u003d1.5.0_09\njava.vendor\u003dSun Microsystems Inc.\nBootLoader constants: OS\u003dwin32, ARCH\u003dx86, WS\u003dwin32, NL\u003den_US\nFramework arguments:  -startup C:\\Eclipse\\eclipse-SDK-I20070323-1616-win32\\eclipse\\plugins\\org.eclipse.equinox.launcher_1.0.0.v20070319.jar\nCommand-line arguments:  -os win32 -ws win32 -arch x86 -startup C:\\Eclipse\\eclipse-SDK-I20070323-1616-win32\\eclipse\\plugins\\org.eclipse.equinox.launcher_1.0.0.v20070319.jar\n\nError\nTue Mar 27 12:19:21 CDT 2007\nException processing async thread queue\n\njava.lang.NullPointerException\nat org.eclipse.jdt.internal.debug.core.model.JDIVoidValue.\u003cinit\u003e(JDIVoidValue.java:26)\nat org.eclipse.jdt.internal.debug.core.model.JDIDebugTarget.voidValue(JDIDebugTarget.java:1762)\nat org.eclipse.jdt.internal.debug.eval.ast.engine.Interpreter.getResult(Interpreter.java:102)\nat org.eclipse.jdt.internal.debug.eval.ast.engine.ASTEvaluationEngine$EvalRunnable.run(ASTEvaluationEngine.java:523)\nat org.eclipse.jdt.internal.debug.core.model.JDIThread$ThreadJob.run(JDIThread.java:2594)\nat org.eclipse.core.internal.jobs.Worker.run(Worker.java:58)\n\nI don\u0027t know what the proper behaviour should be.  It seems pretty strange to want to throw an exception in a detail formatter.  Marking as minor.",
    "\n\n*** This bug has been marked as a duplicate of bug 182158 ***"
  ],
  "commentCreationDates": [
    "2007-03-27T19:31:27+02:00",
    "2007-05-03T18:59:13+02:00"
  ],
  "traces": [
    {
      "exceptionType": "java.lang.NullPointerException",
      "elements": [
        {
          "method": "org.eclipse.jdt.internal.debug.core.model.JDIVoidValue.\u003cinit\u003e",
          "source": "JDIVoidValue.java:26"
        },
        {
          "method": "org.eclipse.jdt.internal.debug.core.model.JDIDebugTarget.voidValue",
          "source": "JDIDebugTarget.java:1762"
        },
        {
          "method": "org.eclipse.jdt.internal.debug.eval.ast.engine.Interpreter.getResult",
          "source": "Interpreter.java:102"
        },
        {
          "method": "org.eclipse.jdt.internal.debug.eval.ast.engine.ASTEvaluationEngine$EvalRunnable.run",
          "source": "ASTEvaluationEngine.java:523"
        },
        {
          "method": "org.eclipse.jdt.internal.debug.core.model.JDIThread$ThreadJob.run",
          "source": "JDIThread.java:2594"
        },
        {
          "method": "org.eclipse.core.internal.jobs.Worker.run",
          "source": "Worker.java:58"
        }
      ],
      "number": 0,
      "commentIndex": 0,
      "bugId": "179571",
      "duplicateId": "182158",
      "date": "2007-03-27T19:31:27+02:00",
      "product": "JDT",
      "component": "Debug",
      "severity": "minor"
    }
  ],
  "groupId": "182158",
  "bugId": "179571",
  "duplicateId": "182158",
  "date": "2007-03-27T19:31:27+02:00",
  "product": "JDT",
  "component": "Debug",
  "severity": "minor"
}